src/libraries/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableDictionary_2.cs

```diff
@@ -142,7 +142,7 @@ public int Count
         /// </value>
         public bool IsEmpty
         {
-            get { return this.Count == 0; }
+            get { return IsEmpty; }
```

Comment:
That seems a bit circular :smile:.

Possibly the analyzer needs to just exclude analysis of `IsEmpty` property getters.